src/Data/Data.ts

```diff
@@ -387,27 +387,39 @@ export function collectCycleData(
     }
     if (config.VERBOSE)
       Logger.mainLogger.debug('Cycle received', cycle.counter, receivedCycleTracker[cycle.counter])
-    
-    const minCycleConfirmations = 
-    Math.min(Math.ceil(NodeList.getActiveNodeCount() / currentConsensusRadius), 5) || 
-    (cycle.counter <= 15 ? 1 : 3);
-    
-
-    for (const value of Object.values(receivedCycleTracker[cycle.counter])) {
-      if (value['saved']) {
-        // If there is a saved cycle, clear the cycleToSave of this counter; This is to prevent saving the another cycle of the same counter
-        for (let i = 0; i < cycleToSave.length; i++) {
-          // eslint-disable-next-line security/detect-object-injection
-          receivedCycleTracker[cycle.counter][cycleToSave[i].marker]['saved'] = false
-        }
-        cycleToSave = []
-        break
+
+    const minCycleConfirmations =
+      Math.min(Math.ceil(NodeList.getActiveNodeCount() / currentConsensusRadius), 5) ||
+      (cycle.counter <= 15 ? 1 : 3);
+
+
+    // Check if any of the markers for this cycle are marekd as saved
+    if (Object.values(receivedCycleTracker[cycle.counter]).some((value) => value['saved'])) {
+      // If there is a saved cycle, clear the cycleToSave of this counter; This is to prevent saving the another cycle of the same counter
+      for (let i = 0; i < cycleToSave.length; i++) {
+        // eslint-disable-next-line security/detect-object-injection
+        receivedCycleTracker[cycle.counter][cycleToSave[i].marker]['saved'] = false
       }
-      if (value['receivedTimes'] >= minCycleConfirmations) {
-        cycleToSave.push(cycle)
-        value['saved'] = true
+      continue
+    }
+
+    // Check if any of the markers for this cycle have received the minimum number of confirmations
+    if (Object.values(receivedCycleTracker[cycle.counter]).some((value) => value['receivedTimes'] >= minCycleConfirmations)) {
+      // Find cycle with max receivedTimes
+      const maxReceivedTimes = Math.max(
```

Comment:
I am not sure if the ask of using most common cycle marker is addressed here.

Reply:
It is. We find the cycle marker received most number of times and use it.